backend/src/api/model/block/mutations.rs

```diff
@@ -0,0 +1,355 @@
+            .query_one(
+                &format!(
+                    "update blocks set
+                        title = case $2::boolean when true then $3 else title end,
+                        text_content = coalesce($4, title)
+                        where id = $1
+                        and type = 'text'
+                        returning {}",
+                    Self::COL_NAMES,
+                ),
+                &[
+                    &id.key_for(Id::BLOCK_KIND)
+                        .ok_or_else(|| invalid_input!("`id` does not refer to a block"))?,
+                    &!set.title.is_implicit_null(),
+                    &set.title.some(),
+                    &set.content,
+                ],
+            )
+            .await?;
+
+        Ok(Self::from_row(updated_block)?)
+    }
+
+    pub(crate) async fn update_series(id: Id, set: UpdateSeriesBlock, context: &Context) -> ApiResult<BlockValue> {
+        let updated_block = context.db(context.require_moderator()?)
+            .query_one(
+                &format!(
+                    "update blocks set
+                        title = case $2::boolean when true then $3 else title end,
+                        series_id = coalesce($4, series_id),
+                        videolist_layout = coalesce($5, videolist_layout),
+                        videolist_order = coalesce($6, videolist_order)
+                        where id = $1
+                        and type = 'series'
+                        returning {}",
+                    Self::COL_NAMES,
+                ),
+                &[
+                    &id.key_for(Id::BLOCK_KIND)
+                        .ok_or_else(|| invalid_input!("`id` does not refer to a block"))?,
+                    &!set.title.is_implicit_null(),
+                    &set.title.some(),
+                    &set.series.map(
+                        |series| series.key_for(Id::SERIES_KIND)
+                            .ok_or_else(|| invalid_input!("`set.series` does not refer to a series"))
+                    ).transpose()?,
+                    &set.layout,
+                    &set.order,
+                ],
+            )
+            .await?;
+
+        Ok(Self::from_row(updated_block)?)
+    }
+
+    pub(crate) async fn remove(id: Id, context: &Context) -> ApiResult<RemovedBlock> {
+        let db = context.db(context.require_moderator()?);
+
+        let result = db
+            .query_one(
+                &format!("delete from blocks
+                    using realms
+                    where blocks.realm_id = realms.id
+                    and blocks.id = $1
+                    returning {}, blocks.index", Realm::col_names("realms")),
+                &[
+                    &id.key_for(Id::BLOCK_KIND)
+                        .ok_or_else(|| invalid_input!("`id` does not refer to a block"))?
+                ],
+            )
+            .await?;
+
+        let realm_key: Key = result.get(0);
```

Comment:
Nope, that's private. :wink:

Reply:
Can't you... make it public? :D 
But ye nevermind, I don't care.

Reply:
Sure, but I figured it was private for a reason. :sweat_smile: 

If you're cool with it, I will gladly make it public, since I agree: this is kind of an ugly workaround.

Reply:
Yes, public then please. Private is just the "default" in a sense, so there is not always a reason behind it.

Reply:
Done in c6b640a.